### PR TITLE
BUGFIX: Undefined variable in AssetController

### PR DESCRIPTION
**Pull request by radmiraal**
The upmerge from 2.2 to 2.3 introduced a regression because of
an undefined variable. This was probably introduced because of
a merge conflict with 427d0ed70e5365fa906f4196c936e4c9271d266e

### CODE REVIEW

File: TYPO3.Media/Classes/TYPO3/Media/Controller/AssetController.php [hunk 1]
```diff
@@ -607,7 +607,7 @@ protected function maximumFileUploadSize()
     public function addFlashMessage($messageBody, $messageTitle = '', $severity = Message::SEVERITY_OK, array $messageArguments = array(), $messageCode = null)
     {
         if (is_string($messageBody)) {
-            $messageBody = $this->translator->translateById($messageBody, $messageArguments, null, null, 'Main', 'TYPO3.Media') ?: $messageBody;
+            $translatedMessageBody = $this->translator->translateById($messageBody, $messageArguments, null, null, 'Main', 'TYPO3.Media') ?: $messageBody;
```
Comment (aertmann) [inline TYPO3.Media/Classes/TYPO3/Media/Controller/AssetController.php]: this is different from 2.2, actually the nested if block shouldn't be there

Comment (radmiraal) [inline TYPO3.Media/Classes/TYPO3/Media/Controller/AssetController.php]: Hey aske, /me confused ;)

Why should this match 2.2? It's a PR to master fixing https://github.com/neos/media/commit/427d0ed70e5365fa906f4196c936e4c9271d266e#diff-49588a4d4cf59c679cf744ce4dc47725 , which is in master right?

**Comment by robertlemke** [comment]
It's yet more complex, and now I also understand why it slipped through in my upmerge:

This bug has been fixed twice.

The first fix is from @johannessteu, but unfortunately only has been pushed to / merged into the 2.3 branch:
https://github.com/neos/media/commit/427d0ed70e5365fa906f4196c936e4c9271d266e#diff-49588a4d4cf59c679cf744ce4dc47725

Then @kitsunet created a similar but a bit different fix, which has been merged into 2.2:
https://github.com/neos/media/commit/7c742c67d3652e60428f93795e0e2a8fa6ec27e9

Very confusing indeed. So, we need to make sure to

- target bugfixes to the correct branch
- double check if the bug has already been fixed in a different PR
- be extra cautious (how extra, extra, extra cautious can one be? ;-) ) with upmerges

**Comment by robertlemke** [comment]
@radmiraal thanks for the patch, but be careful, you targeted it to master and not 2.3. Also, I'm not sure if you had a chance to test it locally?

Anyway, I have adjusted the code to match the one we used in 2.2, so we don't have further problems in upmerges. Also rebased it on 2.3 and force-pushed it. Finally, I changed the base of this pull request to 2.3.

File: TYPO3.Media/Classes/TYPO3/Media/Controller/AssetController.php (hunk 1 above)
Comment (kdambekalns) [inline TYPO3.Media/Classes/TYPO3/Media/Controller/AssetController.php]: … also the ternary should have been removed, this way it creates a useless no-op.

**Comment by kdambekalns** [comment]
@aertmann could you check again and remove your change request?

**Comment by robertlemke** [comment]
merged this as a hotfix, so we can release a new version of 2.3

**Comment by aertmann** [comment]
yep looks good 👍

**Pull request merged**